api/types/events/events.proto

```diff
@@ -1782,6 +1782,16 @@ message RenewableCertificateGenerationMismatch {
         [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
 }
 
+// Unknown is a fallback event used when we don't recognize an event from the backend.
+message Unknown {
+    // Metadata is a common event metadata.
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
```

Comment:
Should we consider adding the code/name of the event in strongly-typed fields?

We could try to populate these on a best-effort basis by attempting to unmarshal the `event` and `code` fields from the JSON data.

The advantage of this is the description in the audit log could then read something like:

`An unknown "foo.bar.baz" (T0123) event`

Reply:
@zmb3 Great idea, added it.